core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -455,14 +466,13 @@ public String resolveIndexRouting(@Nullable String routing, String aliasOrIndex)
             }
             throw new IllegalArgumentException("Alias [" + aliasOrIndex + "] has more than one index associated with it [" + Arrays.toString(indexNames) + "], can't execute a single index op");
         }
-        AliasMetaData aliasMd = alias.getFirstAliasMetaData();
-        if (aliasMd.indexRouting() != null) {
-            if (routing != null) {
-                if (!routing.equals(aliasMd.indexRouting())) {
-                    throw new IllegalArgumentException("Alias [" + aliasOrIndex + "] has index routing associated with it [" + aliasMd.indexRouting() + "], and was provided with routing value [" + routing + "], rejecting operation");
-                }
+        if (routing == null) {
+            AliasMetaData aliasMd = alias.getFirstAliasMetaData();
+            if (aliasMd != null) {
+                routing = aliasMd.indexRouting();
+            } else if (parent != null) {
+                routing = parent;
             }
```

Comment:
IMO if the user adds a routing to an index request this means that he wants to override the default behavior, right? (including the alias routing) otherwise we could have the same reasoning when a parent AND a routing are specified. If the priorities are well defined then I don't think it's a bug but rather a  RTFM ;) .